common/djangoapps/third_party_auth/api/views.py

```diff
@@ -66,34 +182,80 @@ def get(self, request, username):
             JSON serialized list of the providers linked to this user.
 
         """
-        if request.user.username != username:
-            # We are querying permissions for a user other than the current user.
-            if not request.user.is_superuser and not ApiKeyHeaderPermission().has_permission(request, self):
-                # Return a 403 (Unauthorized) without validating 'username', so that we
-                # do not let users probe the existence of other user accounts.
-                return Response(status=status.HTTP_403_FORBIDDEN)
+        identifier = self.get_identifier_for_requested_user(request)
+        return self.do_get(request, identifier)
 
-        try:
-            user = User.objects.get(username=username)
-        except User.DoesNotExist:
-            return Response(status=status.HTTP_404_NOT_FOUND)
+    def get_identifier_for_requested_user(self, _request):
+        """
+        Return an identifier namedtuple for the requested user.
+        """
+        if u'@' in self.kwargs[u'username']:
+            id_kind = u'email'
+        else:
+            id_kind = u'username'
+        return self.identifier(id_kind, self.kwargs[u'username'])
 
-        providers = pipeline.get_provider_user_states(user)
 
-        active_providers = [
-            {
-                "provider_id": assoc.provider.provider_id,
-                "name": assoc.provider.name,
-                "remote_id": assoc.remote_id,
-            }
-            for assoc in providers if assoc.has_account
-        ]
+# TODO: When removing deprecated UserView, rename this view to UserView.
```

Comment:
Do we have an schedule estimate for addressing this tech debt?

Reply:
I think our client would want to have at least one named release that includes the backwards-compatible code, and it could be removed any time after that.